add: Интегральные схемы, пара новых худов

chomasuke opened the pull request:
## Описание
[**Интегральные схемы** - это способ легко автоматизировать различные действия! Вы можете создать схему, которая открывает дверь, когда вы говорите ей секретный код, или схему, которая следует за любым человеком, выкрикивая при этом непристойности, или вообще что угодно!](https://wiki.tgstation13.org/Guide_to_Circuits)

Сопутствующий ущерб:

- Добавлены новые худы для вирусов.
- Наэлектризованные шлюзы отображаются на диагностических худах.
- Порт ТГшных alternate_appearance. Не первой свежести.
- Мини реворк рефлектора. Теперь им можно задать кастомный угол. Спрайты с ТГ.

## Причина создания ПР / Почему это хорошо для игры
[Я когда-то хотел, мне когда-то разрешили](https://discord.com/channels/617003227182792704/755125334097133628/1372118390621081660)￼


## Демонстрация изменений
[Старая демонстрация](https://discord.com/channels/617003227182792704/1014260717949562970/1356764716093014056)
[Очень старая демонстрация](https://discord.com/channels/617003227182792704/1014260717949562970/1353643513891258400)
[Пиздец старая демонстрация](https://discord.com/channels/617003227182792704/1014260717949562970/1350151013453008906)
[Старый документик](https://drive.google.com/file/d/1567Rk44WqePJgQ1LTxz9eyqCJh-o0bBY/view?pli=1)

chomasuke committed:
внатуре фундамент

интегральные схемы, мультитул для схем, имплантер BCI
компоненты для них, кроме /obj/item/circuit_component/wire_bundle, /obj/item/circuit_component/wirenet_receive, /obj/item/circuit_component/wirenet_send и /obj/item/circuit_component/list_literal/wirenet_send
оболочки, кроме /obj/item/shell/scanner_gate, /obj/item/mod/module/circuit, /obj/item/undertile_circuit и /obj/item/wallframe/circuit
компонент /datum/component/usb_port, кроме секурити консоли, фаер алярма и аир алярма
компонент /datum/component/shell
порт рефактора alternate_appearances
небольшой порт рефактора atom_hud
порт худов для вирусов
порт худа наэлектризованного шлюза
порт ПРа #76215 с ТГ (перенос ссылок на ui из сабсистемы в датум)
небольшие изменения радиального меню

dageavtobusnick commented:
Переделай все тгуи компоненты из .js в .tsx, а простые .js файлы в .ts. В js очень тяжело отслеживать и исправлять баги и ошибки из-за динамической типизации, что ведет к неизбежным проблемам.

dageavtobusnick commented on code/game/machinery/lightswitch.dm:
```diff
@@ -169,3 +117,38 @@
+
+/obj/item/circuit_component/light_switch
+	display_name = "Light Switch"
+	desc = "Allows to control the lights of an area."
```
Новый контент нужно локализировать.

dageavtobusnick commented:
Хоть технически у нас и можно использовать jsx файлы, мы их не используем из-за полного перехода на тайпскрипт. Замени jsx на tsx.

chomasuke commented:
> Хоть технически у нас и можно использовать jsx файлы, мы их не используем из-за полного перехода на тайпскрипт. Замени jsx на tsx.

эти коммиты ещё от 27 мая. тупескрипте маринуется в стеше. 
просто убрал конфликты

littleboobs commented:
close?

chomasuke commented:
> close?

ты бы хоть на русском написал, л*кализатор

L343-CJ commented:
интегралки завтра в три?

SaharoK-Er404 committed:
внатуре локалиzация

Настоящий герой на своих титанических плечах вывез локализацию. Два Гл*влока слились, но ОН это сделал! Спасибо, Сизиф. Лучший за работой.

SaharoK-Er404 commented:
интегралки завтра в три

chomasuke committed:
додеп 00

блядский опердрим
блядский тгуй

chomasuke committed:
додеп 01

линтер хуинтер
вынос худа страховки
декленты
худы после ребейза?

chomasuke committed:
додеп 02

рантаймы
обрубок wirenet
почва для остатка wirenet будет отдельно когда-нибудь

chomasuke committed:
додеп 03

снова локализлация
компонент пожарной сигнализации
компонент квантовой платформы
порт реворка отражателя. теперь отражателю можно задать угол
tgui для настройки угла угла отражателя
порт спрайтов отражателя с тг
компонент отражателя. спидран рефлекторной турели?
компонент квантум пада
размеры для заготовок оболочек
ликвидация дублирования для bci худов. возможность удалить все худы bci
бомж-оболочки на карты
линтер хуинтер

в этот додеп должен был войти ещё и компонент для воздушной сигнализации,
но я задушился/заебался/забил-хуй/решил-не-вклинивать-ещё-
(+-)-4000-строк-кода-в-этот-блядский-пр. Будет отдельно когда-нибудь

потраченного времени жаль

chomasuke committed:
чистить вилкой 00

сокращение компонента компакт ремота
стандартизация листов
чуть локализации менюшек

chomasuke committed:
чистить вилкой 01

квантум пад работает
bci труба не сжирает таяр и клоунов
рантаймы

chomasuke committed:
чистить вилкой 02

уменьшение скорости дрона. кд на мув 0.2 -> 0.4 секунды
фикс двойной активации обжект худа на одном атоме
исправление очепятки
экшн HUD_FIVE отображается. изменил название не в .dmi потому что не
верю в свой линукс асепрайт сетап

PlayerUnknown14 committed:
Правки локализации

chomasuke committed:
чистить вилкой 03

оверлеи bci больше не поворачиваются вместе с атомом
nfc/wirenet/ntnet лист литерал работает
компонент пинпоинтер работает
удаление компонента от usb не ломает сам компонент
добавлен забытый адкомпонент для фиреалярма
речарджер впускает в себя дрона
правка правки

PlayerUnknown14 committed:
Доправки локализации (всё кроме TGUI)

chomasuke committed:
чистить вилкой 04

аирлок без электроники доступов успешно собирается
рантайм балуна при сборке оболочки
таргет больше не видит one_person худы
подчистка блядского bci. не последняя
госты могут просматривать плату без оболочки

PlayerUnknown14 committed:
Правки локализации

dageavtobusnick commented on code/controllers/subsystem/tgui.dm:
```diff
@@ -48,7 +46,7 @@ SUBSYSTEM_DEF(tgui)
 			ui.process()
 		else
-			open_uis.Remove(ui)
+			ui.close(0)
```
может ui.close(FALSE)?

dageavtobusnick commented on code/controllers/subsystem/processing/instruments.dm:
```diff
@@ -22,6 +22,7 @@ PROCESSING_SUBSYSTEM_DEF(instruments)
 	var/static/current_instrument_channels = 0
 	/// Single cached list for synthesizer instrument ids, so you don't have to have a new list with every synthesizer.
 	var/static/list/synthesizer_instrument_ids
+	var/static/list/note_sustain_modes = list("Линейно" = SUSTAIN_LINEAR, "Экспоненциально" = SUSTAIN_EXPONENTIAL)
```
Если ключи где-то прописываются в коде, например в качестве значений переменных, то лучше их дефайнизировать. Если ключи просто для выбора игроком и негде больше они не используются, то не нужно. Это нужно чтобы минимизировать ошибки в проверках.

chomasuke replied:
эти ключи только для игрока

dageavtobusnick commented on code/controllers/subsystem/tgui.dm:
```diff
@@ -187,15 +184,14 @@ SUBSYSTEM_DEF(tgui)
-		return count
-	for(var/datum/tgui/ui in open_uis_by_src[key])
+	if(!LAZYLEN(src_object.open_uis))
+		return 0
```
пустую строку после if

dageavtobusnick commented on code/controllers/subsystem/tgui.dm:
```diff
@@ -209,17 +205,16 @@ SUBSYSTEM_DEF(tgui)
-	if(isnull(open_uis_by_src[key]) || !islist(open_uis_by_src[key]))
-		return
-	for(var/datum/tgui/ui in open_uis_by_src[key])
+	if(!LAZYLEN(src_object.open_uis))
```
аналогично

dageavtobusnick commented on code/datums/components/shell.dm:
```diff
@@ -0,0 +1,409 @@
+		return
+	if(attached_circuit.admin_only)
+		return
+	if(shell_flags & SHELL_FLAG_CIRCUIT_UNREMOVABLE)
```
раздели ифы пустыми строками

dageavtobusnick commented on code/datums/components/shell.dm:
```diff
@@ -0,0 +1,409 @@
+	if(!is_authorized(attacker))
+		return
+
+	if(istype(item, /obj/item/stock_parts/cell))
```
вроде бы есть isхелпер для батареек

dageavtobusnick commented on code/datums/components/shell.dm:
```diff
@@ -0,0 +1,409 @@
+			attached_circuit.add_component_manually(item, attacker)
+			return COMPONENT_CANCEL_ATTACK_CHAIN
+
+	if(!istype(item, /obj/item/integrated_circuit))
```
Стоит сделать isхелпер и занинить на него. Думаю такая проверка много где используется

dageavtobusnick commented on code/datums/components/shell.dm:
```diff
@@ -0,0 +1,409 @@
+			attached_circuit.owner_id = WEAKREF(item)
+			return COMPONENT_CANCEL_ATTACK_CHAIN
+
+		if(istype(item, /obj/item/circuit_component))
```
аналогично

dageavtobusnick commented on code/datums/components/shell.dm:
```diff
@@ -0,0 +1,409 @@
+	if(!attached_circuit)
+		return
+
+	if(!istype(tool, /obj/item/multitool/circuit))
```
изхелпер?

dageavtobusnick commented on code/datums/components/shell.dm:
```diff
@@ -0,0 +1,409 @@
+ */
+/datum/component/shell/proc/on_circuit_moved(obj/item/integrated_circuit/circuit, atom/old_loc)
+	SIGNAL_HANDLER
+	if(circuit.loc != parent)
```
ранний ретурн

dageavtobusnick commented on code/datums/components/shell.dm:
```diff
@@ -0,0 +1,409 @@
+		source.balloon_alert(user, "закрыто!")
+		return COMPONENT_CANCEL_ADD_COMPONENT
+
+	if(attached_circuit.current_size + added_comp.circuit_size > capacity)
```
ранний ретурн

dageavtobusnick commented on code/datums/components/shell.dm:
```diff
@@ -0,0 +1,409 @@
+	SEND_SIGNAL(src, COMSIG_SHELL_CIRCUIT_ATTACHED)
+	if(!(shell_flags & SHELL_FLAG_CIRCUIT_UNREMOVABLE) && !circuitboard.admin_only)
+		RegisterSignal(circuitboard, COMSIG_MOVABLE_MOVED, PROC_REF(on_circuit_moved))
+	if(shell_flags & SHELL_FLAG_REQUIRE_ANCHOR)
```
во всем проке отдели if-ы пустыми строками, где этого еще нету

dageavtobusnick commented on code/datums/components/usb_port.dm:
```diff
@@ -0,0 +1,251 @@
+		RegisterSignal(component, COMSIG_CIRCUIT_COMPONENT_SAVE, PROC_REF(save_component))
+		circuit_components += component
+
+	if(should_register)
```
ранний ретурн

dageavtobusnick commented on code/datums/components/usb_port.dm:
```diff
@@ -0,0 +1,251 @@
+		if(component.type in components_in_list)
+			continue
+		components += component.type
+	set_circuit_components(components)
```
пустую строку после for

dageavtobusnick commented on code/datums/components/usb_port.dm:
```diff
@@ -0,0 +1,251 @@
+
+	if(isnull(attached_circuit))
+		examine_text += span_notice("На передней панели расположен USB-порт.")
+	else
```
сделай с ретурном и без else

dageavtobusnick commented on code/datums/diseases/viruses/advance/advance.dm:
```diff
@@ -217,17 +217,19 @@ GLOBAL_LIST_EMPTY(archive_diseases)
+			if(5)
+				severity = DISEASE_SEVERITY_DANGEROUS
+			if(6 to INFINITY)
+				severity = DISEASE_SEVERITY_BIOHAZARD
```
Как минимум сделай хелпер и перенеси его к get_disease_severity_value. Иначе добавление нового уровня превратится в ад

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,145 +1,194 @@
-*/
-/atom/proc/display_alt_appearance(key, list/displayTo)
-	if(!alternate_appearances || !key)
+		for(var/K in alternate_appearances)
```
имя переменной

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,145 +1,194 @@
-/atom/proc/display_alt_appearance(key, list/displayTo)
-	if(!alternate_appearances || !key)
+		for(var/K in alternate_appearances)
+			var/datum/atom_hud/alternate_appearance/AA = alternate_appearances[K]
```
имя переменной

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,145 +1,194 @@
+/datum/atom_hud/alternate_appearance/proc/check_hud(mob/source)
+	SIGNAL_HANDLER
+	// Attempt to re-apply the hud entirely
+	if(!apply_to_new_mob(source))
```
ранний ретурн

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,145 +1,194 @@
+
+/datum/atom_hud/alternate_appearance/add_atom_to_hud(atom/A, image/I)
+	. = ..()
+	if(.)
```
ранний ретурн

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,145 +1,194 @@
+
+/datum/atom_hud/alternate_appearance/remove_atom_from_hud(atom/A)
+	. = ..()
+	if(.)
```
ранний ретурн

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,145 +1,194 @@
+
+	if((options & AA_TARGET_SEE_APPEARANCE) && ismob(target))
+		show_to(target)
+	if(add_ghost_version)
```
ранний ретурн и пустая строка между if

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,145 +1,194 @@
+	image = I
+	target = I.loc
+	LAZYADD(target.update_on_z, image)
+	if(transfer_overlays)
```
пустая строка над if

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,145 +1,194 @@
+	return FALSE
+
+/// Checks if the passed mob should be seeing this hud
+/datum/atom_hud/alternate_appearance/proc/mobShouldSee(mob/M)
```
имя переменной и прока

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,145 +1,194 @@
+		// If that failed, probably shouldn't be seeing it at all, so nuke it
+		hide_from(source, absolute = TRUE)
+
+/datum/atom_hud/alternate_appearance/add_atom_to_hud(atom/A, image/I)
```
Имена переменных

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,145 +1,194 @@
+		LAZYINITLIST(A.alternate_appearances)
+		A.alternate_appearances[appearance_key] = src
+
+/datum/atom_hud/alternate_appearance/remove_atom_from_hud(atom/A)
```
имя переменной

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,145 +1,194 @@
+	var/datum/atom_hud/alternate_appearance/basic/observers/ghost_appearance
+	uses_global_hud_category = FALSE
+
+/datum/atom_hud/alternate_appearance/basic/New(key, image/I, options = AA_TARGET_SEE_APPEARANCE)
```
Имя переменной

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,145 +1,194 @@
+		ghost_image.alpha = 128
+		ghost_appearance = new /datum/atom_hud/alternate_appearance/basic/observers(key + "_observer", ghost_image, NONE)
+
+/datum/atom_hud/alternate_appearance/basic/Destroy()
```
```suggestion
/datum/atom_hud/alternate_appearance/basic/Destroy(force)
```

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,145 +1,194 @@
+	LAZYREMOVE(target.update_on_z, image)
+	QDEL_NULL(image)
+	target = null
+	if(ghost_appearance)
```
ранний ретурн

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,145 +1,194 @@
+		COMSIG_BODY_TRANSFER_TO,
+	))
+
+/datum/atom_hud/alternate_appearance/basic/add_atom_to_hud(atom/A)
```
имя переменной

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,145 +1,194 @@
+	A.hud_list[appearance_key] = image
+	. = ..()
+
+/datum/atom_hud/alternate_appearance/basic/remove_atom_from_hud(atom/A)
```
имя переменной

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,145 +1,194 @@
+/datum/atom_hud/alternate_appearance/basic/everyone
+	add_ghost_version = TRUE
+
+/datum/atom_hud/alternate_appearance/basic/everyone/mobShouldSee(mob/M)
```
имя переменной и прока

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,145 +1,194 @@
+
+/datum/atom_hud/alternate_appearance/basic/silicons
+
+/datum/atom_hud/alternate_appearance/basic/silicons/mobShouldSee(mob/M)
```
аналогично

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,145 +1,194 @@
+
+/datum/atom_hud/alternate_appearance/basic/observers
+
+/datum/atom_hud/alternate_appearance/basic/observers/mobShouldSee(mob/M)
```
аналогично

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,145 +1,194 @@
+	/// The guy who gets to see the image
+	var/mob/seer
+
+/datum/atom_hud/alternate_appearance/basic/one_person/mobShouldSee(mob/M)
```
аналогично

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,145 +1,194 @@
+/// Shows the image to everyone but one person
+/datum/atom_hud/alternate_appearance/basic/one_person/reversed
 
+/datum/atom_hud/alternate_appearance/basic/one_person/reversed/mobShouldSee(mob/M)
```
аналогично

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,145 +1,194 @@
+/datum/atom_hud/alternate_appearance/basic/one_person/mobShouldSee(mob/M)
+	return M == seer
+
+/datum/atom_hud/alternate_appearance/basic/one_person/New(key, image/I, options = NONE, mob/living/seer)
```
имя переменной

dageavtobusnick reviewed (changes requested):
Часть 1

dageavtobusnick commented on code/game/data_huds.dm:
```diff
@@ -119,7 +120,24 @@
+	var/severity
+
+	for(var/thing in diseases)
+		var/datum/disease/D = thing
```
правильнее использовать as anything in

dageavtobusnick commented on code/game/data_huds.dm:
```diff
@@ -299,84 +313,110 @@
+			set_hud_image_state(STATUS_HUD, "hudill0")
+
+		if(DISEASE_SEVERITY_POSITIVE)
+			set_hud_image_state(STATUS_HUD, "hudbuff")
```
ммм. Учитывая, что тут уже 3 связи (DISEASE_SEVERITY имя DISEASE_SEVERITY число и икон стейт), лучше будет использовать датумы и их глобальный список для установления связи, заместо кучи костыльных хелперов.

dageavtobusnick commented on code/game/data_huds.dm:
```diff
@@ -660,6 +696,42 @@
+	if(!istype(holder)) // Can contain lists for HUD_LIST_LIST hinted HUDs, if someone fucks up and passes this here we wanna know about it
+		CRASH("[src] ([type]) had a HUD_LIST_LIST hud_type [hud_type] passed into set_hud_image_state!")
+	holder.icon_state = hud_state
+	if(x_offset || y_offset)
```
ранний ретурн

dageavtobusnick commented on code/game/data_huds.dm:
```diff
@@ -660,6 +696,42 @@
+	var/image/holder = hud_list[hud_type]
+	if(!holder)
+		return
+	if(!istype(holder)) // Can contain lists for HUD_LIST_LIST hinted HUDs, if someone fucks up and passes this here we wanna know about it
```
пустые строки между ифами и остальным кодом

dageavtobusnick commented on code/game/jobs/access.dm:
```diff
@@ -1,16 +1,22 @@
+	var/result_bitflags = SEND_SIGNAL(src, COMSIG_OBJ_ALLOWED, accessor)
+	if(result_bitflags & COMPONENT_OBJ_ALLOW)
+		return TRUE
+	if(result_bitflags & COMPONENT_OBJ_DISALLOW) // override all other checks
```
пустые строки между ифами и остальным кодом

dageavtobusnick commented on code/game/machinery/computer/crew.dm:
```diff
@@ -51,3 +66,73 @@
+
+/obj/item/circuit_component/medical_console_data/register_usb_parent(atom/movable/shell)
+	. = ..()
+	if(istype(shell, /obj/machinery/computer/crew))
```
ранний ретурн

dageavtobusnick commented on code/game/machinery/computer/crew.dm:
```diff
@@ -51,3 +66,73 @@
+	if(!attached_console || !GLOB.crew_repository)
+		return
+
+	var/turf/T = get_turf(attached_console)
```
имя переменной

dageavtobusnick commented on code/game/machinery/doors/airlock.dm:
```diff
@@ -1418,20 +1431,28 @@ GLOBAL_LIST_EMPTY(airlock_emissive_underlays)
+/obj/machinery/door/airlock/proc/set_bolt(should_bolt)
+	if(locked == should_bolt)
+		return
+	SEND_SIGNAL(src, COMSIG_AIRLOCK_SET_BOLT, should_bolt)
```
пустую строку после if

dageavtobusnick commented on code/game/machinery/firealarm.dm:
```diff
@@ -361,6 +369,61 @@ Just a object used in constructing fire alarms
+
+/obj/item/circuit_component/firealarm/register_usb_parent(atom/movable/parent)
+	. = ..()
+	if(istype(parent, /obj/machinery/firealarm))
```
ранний ретурн + изхелпер, если он есть в коде

dageavtobusnick commented on code/game/machinery/firealarm.dm:
```diff
@@ -361,6 +369,61 @@ Just a object used in constructing fire alarms
+
+/obj/item/circuit_component/firealarm/proc/on_firealarm_triggered(datum/source)
+	SIGNAL_HANDLER
+	is_on.set_output(1)
```
is_on.set_output(TRUE) ?

dageavtobusnick commented on code/game/machinery/firealarm.dm:
```diff
@@ -361,6 +369,61 @@ Just a object used in constructing fire alarms
+
+/obj/item/circuit_component/firealarm/proc/on_firealarm_reset(datum/source)
+	SIGNAL_HANDLER
+	is_on.set_output(0)
```
аналогично, но FALSE?

dageavtobusnick commented on code/game/machinery/lightswitch.dm:
```diff
@@ -28,41 +25,27 @@
+		area = text2path(area)
+	if(ispath(area))
+		area = GLOB.areas_by_type[area]
+	if(!area)
```
пустые строки

dageavtobusnick commented on code/game/machinery/lightswitch.dm:
```diff
@@ -169,3 +117,38 @@
+
+/obj/item/circuit_component/light_switch/register_usb_parent(atom/movable/parent)
+	. = ..()
+	if(istype(parent, /obj/machinery/light_switch))
```
ранний ретурн и изхелпер если есть в коде, либо если такая проверка встречается не единажды

dageavtobusnick commented on code/game/machinery/quantum_pad.dm:
```diff
@@ -114,53 +119,52 @@
+	if(stat & NOPOWER)
+		to_chat(user, span_warning("[src] is unpowered!"))
+		return
+	if(!target_pad || QDELETED(target_pad) || target_pad.stat & NOPOWER)
```
пустую строку

dageavtobusnick commented on code/game/machinery/quantum_pad.dm:
```diff
@@ -114,53 +119,52 @@
+	flick("[initial(target_pad.icon_state)]-beam", target_pad)
+	playsound(get_turf(target_pad), 'sound/weapons/emitter2.ogg', 25, TRUE)
+	var/tele_success = TRUE
+	for(var/atom/movable/ROI in get_turf(src))
```
имя переменной

dageavtobusnick commented on code/game/machinery/quantum_pad.dm:
```diff
@@ -114,53 +119,52 @@
+		// if is anchored, don't let through
+		if(ROI.anchored)
+			if(isliving(ROI))
+				var/mob/living/L = ROI
```
имя переменной

dageavtobusnick commented on code/game/machinery/quantum_pad.dm:
```diff
@@ -114,53 +119,52 @@
-				to_chat(user, span_warning("Teleport failed due to bluespace interference."))
+				else
+					continue
+			else if(!isobserver(ROI))
```
попробуй использовать ранние continue, чтобы разбить это безобразие

dageavtobusnick commented on code/game/objects/effects/info.dm:
```diff
@@ -0,0 +1,25 @@
+/obj/effect/info/Initialize(mapload, info_text)
+	. = ..()
+
+	if(!isnull(info_text))
```
ранний ретурн

dageavtobusnick commented on code/game/objects/items/weapons/cash.dm:
```diff
@@ -58,6 +58,9 @@
 	. = ..()
 
+/obj/item/stack/spacecash/get_item_credit_value()
+	return (amount)
```
зачем скобки?

dageavtobusnick commented on code/game/objects/objs.dm:
```diff
@@ -208,7 +208,8 @@
 
-/obj/proc/hear_talk(mob/M, list/message_pieces)
+/obj/proc/hear_talk(mob/speaker, list/message_pieces)
+	SEND_SIGNAL(src, COMSIG_MOVABLE_HEAR, speaker, message_pieces)
```
Сделай SHOULD_CALL_PARENT

dageavtobusnick commented on code/game/objects/structures/reflector.dm:
```diff
@@ -1,48 +1,144 @@
+	playsound(user, 'sound/items/Ratchet.ogg', 50, TRUE)
+	TOOL_DISMANTLE_SUCCESS_MESSAGE
+	new framebuildstacktype(drop_location(), framebuildstackamount)
+	if(buildstackamount)
```
пустая строка

dageavtobusnick commented on code/game/objects/structures/reflector.dm:
```diff
@@ -1,48 +1,144 @@
-/obj/structure/reflector/attackby(obj/item/I, mob/user, params)
+/obj/structure/reflector/wrench_act(mob/user, obj/item/I)
+	. = TRUE
+	if(anchored)
```
пустая строка

dageavtobusnick commented on code/game/objects/structures/reflector.dm:
```diff
@@ -1,48 +1,144 @@
+	else
+		WELDER_ATTEMPT_FLOOR_WELD_MESSAGE
+		set_anchored(TRUE)
+	WELDER_FLOOR_WELD_SUCCESS_MESSAGE
```
пустая строка

dageavtobusnick commented on code/game/objects/structures/reflector.dm:
```diff
@@ -83,111 +171,175 @@
+/obj/structure/reflector/singularity_act()
+	if(admin)
+		return
+	else
```
А зачем тут else?

dageavtobusnick commented on code/game/objects/structures/reflector.dm:
```diff
@@ -83,111 +171,175 @@
+
+/obj/item/circuit_component/reflector/register_usb_parent(atom/movable/parent)
+	. = ..()
+	if(istype(parent, /obj/structure/reflector))
```
ранний ретурн

dageavtobusnick commented on code/game/objects/structures/reflector.dm:
```diff
@@ -83,111 +171,175 @@
+		ui?.close()
+		return
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
```
ранний ретурн

dageavtobusnick commented on code/game/objects/structures/reflector.dm:
```diff
@@ -83,111 +171,175 @@
+		user.balloon_alert(user, "nothing to rotate!")
+		return
+	if(!can_rotate)
+		user.balloon_alert(user, "can't rotate!")
```
перевод ебалунов?

dageavtobusnick commented on code/game/objects/structures/reflector.dm:
```diff
@@ -83,111 +171,175 @@
+			var/new_angle = rotation_angle + params["rotation_angle"]
+			if(!isnull(new_angle))
+				set_reflector_angle(SIMPLIFY_DEGREES(new_angle))
+			return TRUE
```
отделить ифы пустыми строками

dageavtobusnick commented on code/game/objects/structures/reflector.dm:
```diff
@@ -83,111 +171,175 @@
+
+/obj/structure/reflector/wrenched/Initialize(mapload)
+	. = ..()
+
```
зачем тут пустая строка?

dageavtobusnick commented on code/modules/atmospherics/machinery/components/binary_devices/pump.dm:
```diff
@@ -234,3 +240,103 @@ Thus, the two variables affect pump operation are set in New():
+
+/obj/item/circuit_component/atmos_pump/register_usb_parent(atom/movable/shell)
+	. = ..()
+	if(istype(shell, /obj/machinery/atmospherics/binary/pump))
```
ранний ретутн

dageavtobusnick commented on code/modules/atmospherics/machinery/components/binary_devices/pump.dm:
```diff
@@ -234,3 +240,103 @@ Thus, the two variables affect pump operation are set in New():
+	CIRCUIT_TRIGGER
+	if(!connected_pump)
+		return
+	var/datum/gas_mixture/air_input = connected_pump.air1
```
ифы лучше отделить пустыми строками во всем файле

dageavtobusnick commented on code/modules/atmospherics/machinery/components/binary_devices/valve.dm:
```diff
@@ -142,3 +148,58 @@
+
+/obj/item/circuit_component/digital_valve/register_usb_parent(atom/movable/shell)
+	. = ..()
+	if(istype(shell, /obj/machinery/atmospherics/binary/valve/digital))
```
ранний ретурн

dageavtobusnick commented on code/modules/atmospherics/machinery/components/binary_devices/valve.dm:
```diff
@@ -142,3 +148,58 @@
+
+	if(COMPONENT_TRIGGERED_BY(open, port) && !attached_valve.on)
+		attached_valve.toggle()
+	if(COMPONENT_TRIGGERED_BY(close, port) && attached_valve.on)
```
пустые строки

dageavtobusnick commented on code/modules/atmospherics/machinery/components/binary_devices/valve.dm:
```diff
@@ -142,3 +148,58 @@
+		closed.set_output(COMPONENT_SIGNAL)
+
+/obj/item/circuit_component/digital_valve/input_received(datum/port/input/port)
+
```
зачем тут пустая строка?

dageavtobusnick commented on code/modules/atmospherics/machinery/components/binary_devices/volume_pump.dm:
```diff
@@ -228,3 +234,102 @@ Thus, the two variables affect pump operation are set in New():
+
+/obj/item/circuit_component/atmos_volume_pump/register_usb_parent(atom/movable/shell)
+	. = ..()
+	if(istype(shell, /obj/machinery/atmospherics/binary/volume_pump))
```
ранний ретурн

dageavtobusnick commented on code/modules/atmospherics/machinery/components/binary_devices/volume_pump.dm:
```diff
@@ -228,3 +234,102 @@ Thus, the two variables affect pump operation are set in New():
+	var/datum/gas_mixture/air_output = connected_pump.air2
+	input_pressure.set_output(air_input.return_pressure())
+	output_pressure.set_output(air_output.return_pressure())
+	input_temperature.set_output(air_input.return_temperature())
```
Та же проблема с пустыми строками, что и в прошлом файле. ифы, свичи и циклы лучше отделять пустой строкой от остального кода. Если только прок не начинается с этой конструкции. В начеле прока отступ не нужен. Поправь во всем коде. Больше указывать на это не буду в основном.

dageavtobusnick reviewed (changes requested):
Часть 2

dageavtobusnick commented on code/modules/atmospherics/machinery/other/meter.dm:
```diff
@@ -138,3 +143,41 @@
+
+/obj/item/circuit_component/atmos_meter/register_usb_parent(atom/movable/shell)
+	. = ..()
+	if(istype(shell, /obj/machinery/atmospherics/meter))
```
ранний ретурн

dageavtobusnick commented on code/modules/mob/hear_say.dm:
```diff
@@ -90,11 +90,11 @@
 	return "[verb]: \"[message]\""
 
 /mob/proc/hear_say(list/message_pieces, verb = "говор%(ит,ят)%", italics = FALSE, mob/speaker = null, sound/speech_sound, sound_vol, sound_frequency, use_voice = TRUE, is_whisper = FALSE)
+	SEND_SIGNAL(src, COMSIG_MOVABLE_HEAR, speaker, message_pieces)
```
Уверен, что сигнал будет отправлен у всех мобов? Лучше сделать шуд кол парант

chomasuke replied:
кому понадобится, тот получит. кроме головной боли, кол парент в этом проке ничего не даст

dageavtobusnick commented on code/modules/paperwork/photography/camera.dm:
```diff
@@ -192,10 +197,10 @@
+	var/range = picture_size * 2 + 1
 	var/clone_area = SSmapping.request_turf_block_reservation(range, range, 1)
-	var/icon/photoimage = camera_get_icon(turfs, target, user, size*32, clone_area, size, range)
+	var/icon/photoimage = camera_get_icon(turfs, target, user, picture_size*32, clone_area, picture_size, range)
```
```suggestion
	var/icon/photoimage = camera_get_icon(turfs, target, user, picture_size * ICON_SIZE_ALL, clone_area, picture_size, range)
```

dageavtobusnick commented on code/modules/power/lighting.dm:
```diff
@@ -834,9 +834,11 @@
+	addtimer(CALLBACK(src, PROC_REF(actually_explode)), 2)
+
+/obj/machinery/light/proc/actually_explode()
+	var/turf/T = get_turf(loc)
```
имя переменной

dageavtobusnick commented on code/modules/recycling/conveyor2.dm:
```diff
@@ -735,6 +738,60 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
+	if(istype(shell, /obj/machinery/conveyor_switch))
+		attached_switch = shell
+		if(!attached_switch.one_way)
+			reverse = add_input_port("Реверс", PORT_TYPE_SIGNAL, trigger = PROC_REF(reverse))
```
ранние ретурны

dageavtobusnick commented on code/modules/research/circuitprinter.dm:
```diff
@@ -90,18 +98,50 @@ using metal and glass, it uses glass and reagents (usually sulfuric acis).
+		return
+
+	var/datum/research/research_console = linked_console.files
+	for(var/v in research_console.known_designs)
```
имя переменной

dageavtobusnick commented on code/modules/research/circuitprinter.dm:
```diff
@@ -90,18 +98,50 @@ using metal and glass, it uses glass and reagents (usually sulfuric acis).
+
+/obj/machinery/r_n_d/circuit_imprinter/ui_interact(mob/user, datum/tgui/ui = null)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
```
ранний ретурн

dageavtobusnick commented on code/modules/research/circuitprinter.dm:
```diff
@@ -90,18 +98,50 @@ using metal and glass, it uses glass and reagents (usually sulfuric acis).
+	return interact(user)
+
+/obj/machinery/r_n_d/circuit_imprinter/interact(mob/user)
+	if(!disabled)
```
ранний ретурн

dageavtobusnick commented on code/modules/research/circuitprinter.dm:
```diff
@@ -110,33 +150,282 @@ using metal and glass, it uses glass and reagents (usually sulfuric acis).
+	if(amount_inserted)
+		qdel(tool)
+		to_chat(user, span_notice("[capitalize(declent_ru(NOMINATIVE))] переребатывает [tool.declent_ru(ACCUSATIVE)] в [amount_inserted /  SHEET_VOLUME] единиц[DECL_SEC_MIN(amount_inserted /  SHEET_VOLUME)] материала."))
+	else
```
реверт и ранний ретурн

dageavtobusnick commented on code/modules/research/circuitprinter.dm:
```diff
@@ -110,33 +150,282 @@ using metal and glass, it uses glass and reagents (usually sulfuric acis).
+		return FALSE
+
+	switch(choice)
+		if("Link Circuit")
```
дефайнизировать локально. Так же по идее нужно перевести.

dageavtobusnick commented on code/modules/research/circuitprinter.dm:
```diff
@@ -110,33 +150,282 @@ using metal and glass, it uses glass and reagents (usually sulfuric acis).
+	if(!(design?.build_type & IMPRINTER))
+		return
+
+	if(try_use_materials(design.materials))
```
ранний ретурн

dageavtobusnick commented on code/modules/research/designs/wiremod_designs.dm:
```diff
@@ -0,0 +1,748 @@
+
+/datum/design/component/New()
+	. = ..()
+	if(build_path)
```
ранний ретурн

dageavtobusnick commented on code/modules/research/designs/wiremod_designs.dm:
```diff
@@ -0,0 +1,748 @@
+
+/datum/design/assembly_shell
+	name = "Assembly Shell"
+	desc = "An assembly shell that can be attached to wires and other assemblies."
```
По идее как минимум описания нужно перевести.

chomasuke replied:
уже не нужно

dageavtobusnick commented on code/modules/wiremod/components/abstract/module.dm:
```diff
@@ -0,0 +1,336 @@
+	if(ispath(type, /obj/item/circuit_component/module_input))
+		return attached_module.input_component
+
+	if(ispath(type, /obj/item/circuit_component/module_output))
```
изхелперы?

dageavtobusnick commented on code/modules/wiremod/components/abstract/module.dm:
```diff
@@ -0,0 +1,336 @@
+		balloon_alert(user, "не подходит для модуля!")
+		return
+	. = ..()
+	if(attached_module)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/abstract/module.dm:
```diff
@@ -0,0 +1,336 @@
+					return
+				component_port.set_datatype(type)
+				internal_component_port.set_datatype(type)
+			else
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/abstract/variable.dm:
```diff
@@ -0,0 +1,64 @@
+
+/obj/item/circuit_component/variable/proc/remove_current_variable()
+	SIGNAL_HANDLER
+	if(current_variable)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/action/mmi.dm:
```diff
@@ -0,0 +1,180 @@
+
+	if(COMPONENT_TRIGGERED_BY(eject, port))
+		remove_current_brain()
+	if(COMPONENT_TRIGGERED_BY(send, port))
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/action/mmi.dm:
```diff
@@ -0,0 +1,180 @@
+	return ..()
+
+/obj/item/circuit_component/mmi/input_received(datum/port/input/port)
+
```
зачем тут пустая строка?

dageavtobusnick commented on code/modules/wiremod/components/action/mmi.dm:
```diff
@@ -0,0 +1,180 @@
+
+/obj/item/circuit_component/mmi/proc/handle_attack_by(atom/movable/shell, obj/item/item, mob/living/attacker)
+	SIGNAL_HANDLER
+	if(istype(item, /obj/item/mmi))
```
ранний ретурн и изхелпер

dageavtobusnick commented on code/modules/wiremod/components/action/mmi.dm:
```diff
@@ -0,0 +1,180 @@
+/obj/item/circuit_component/mmi/proc/mmi_moved(atom/movable/mmi)
+	SIGNAL_HANDLER
+
+	if(mmi.loc != src)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/action/mmi.dm:
```diff
@@ -0,0 +1,180 @@
+	if(old_mmi)
+		old_mmi.remote_control = null
+		UnregisterSignal(old_mmi, COMSIG_MOB_CLICKON)
+	if(new_mmi)
```
ранний ретурн

chomasuke replied:
не стоит

dageavtobusnick replied:
стоит

dageavtobusnick commented on code/modules/wiremod/components/action/mmi.dm:
```diff
@@ -0,0 +1,180 @@
+	if(modifiers[ALT_CLICK])
+		clicked_atom.set_output(target)
+		alt_attack.set_output(COMPONENT_SIGNAL)
+		. = COMSIG_MOB_CANCEL_CLICKON
```
ранний ретурн тут и ниже

dageavtobusnick commented on code/modules/wiremod/components/action/pathfind.dm:
```diff
@@ -0,0 +1,115 @@
+		else
+			TIMER_COOLDOWN_START(parent, COOLDOWN_CIRCUIT_PATHFIND_DIF, different_path_cooldown)
+			next_turf = get_turf(path[1])
+			output.set_output(next_turf)
```
слишком большой прок, разбей на несколько. Возможно получится что-то обопщить.

dageavtobusnick commented on code/modules/wiremod/components/action/radio.dm:
```diff
@@ -0,0 +1,122 @@
+/obj/item/circuit_component/radio/register_shell(atom/movable/shell)
+	parent_shell = shell
+	var/potential_fingerprints = shell.fingerprintslast
+	if(!isnull(potential_fingerprints))
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/action/radio.dm:
```diff
@@ -0,0 +1,122 @@
+	trigger_input = add_input_port("Вызов", PORT_TYPE_SIGNAL)
+	trigger_output = add_output_port("Вызвано", PORT_TYPE_SIGNAL)
+
+/obj/item/circuit_component/radio/Destroy()
```
```suggestion
/obj/item/circuit_component/radio/Destroy(force)
```

dageavtobusnick commented on code/modules/wiremod/components/action/radio.dm:
```diff
@@ -0,0 +1,122 @@
+		radio_connection = SSradio.add_object(src, frequency, RADIO_SIGNALER)
+		current_freq = frequency
+
+	if(COMPONENT_TRIGGERED_BY(trigger_input, port))
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/action/soundemitter.dm:
```diff
@@ -0,0 +1,95 @@
+	if(!sound_to_play)
+		return
+
+	var/actual_frequency = 1 + (frequency.value/100)
```
```suggestion
	var/actual_frequency = 1 + (frequency.value / 100)
```

dageavtobusnick commented on code/modules/wiremod/components/action/soundemitter.dm:
```diff
@@ -0,0 +1,95 @@
+		return
+
+	var/actual_frequency = 1 + (frequency.value/100)
+	var/actual_volume = max_volume * (volume.value/100)
```
```suggestion
	var/actual_volume = max_volume * (volume.value / 100)
```

dageavtobusnick commented on tgui/packages/tgui/interfaces/AirlockElectronics.tsx:
```diff
@@ -108,12 +110,22 @@ const UnrestrictedAccess = (props: unknown) => {
+                on: !shell,
+              });
+            }}
+            tooltip="Whether this airlock can have an integrated circuit placed inside of it or not."
```
перевод?

dageavtobusnick commented on tgui/packages/tgui/interfaces/CircuitAccessChecker.tsx:
```diff
@@ -0,0 +1,54 @@
+          <LabeledList.Item label="Access Required">
+            <Button
+              icon={oneAccess ? 'unlock' : 'lock'}
+              content={oneAccess ? 'One' : 'All'}
```
контент является устаревшим вариантом. используй children(Тоесть помещай содержимое по такому принципу <Button>содержимое</Button>. Поправь во всем своем тгуи. Это касается всех <Button>

chomasuke replied:
может он и устарел, но поставленную задачу - рисовать символьцы на кнопочке - он выполняет. какой выхлоп с замены, если чилдрена ещё и оборачивать нужно?

dageavtobusnick replied:
Единый стиль. Большая часть кнопок в билде(90-95%) переделана под новый формат. Кроме того, устаревшим он явояется и в tgui core, а это значит, что когда-нить он вполне может быть убран оттуда.

dageavtobusnick commented on tgui/packages/tgui/interfaces/IntegratedCircuit/index.tsx:
```diff
@@ -0,0 +1,646 @@
+  // mouse up called whilst over a port. This means we can check if selectedPort
+  // exists and do perform some actions if it does.
+  handlePortUp(portIndex, componentId, port, isOutput, event) {
+    // Вот этот пидорас срёт
```
кхм

dageavtobusnick commented on code/modules/wiremod/components/action/speech.dm:
```diff
@@ -0,0 +1,35 @@
+	if(TIMER_COOLDOWN_RUNNING(parent.shell, COOLDOWN_CIRCUIT_SPEECH))
+		return
+
+	if(message.value)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/atom/pinpointer.dm:
```diff
@@ -0,0 +1,58 @@
+		x_pos.set_output(location?.x)
+		y_pos.set_output(location?.y)
+		z_pos.set_output(location?.z)
+	else
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/atom/remotecam.dm:
```diff
@@ -0,0 +1,438 @@
+		start_process()
+		cameranet_add()
+		current_camera_state = TRUE
+	else if(COMPONENT_TRIGGERED_BY(stop, port))
```
ранний ретрун

dageavtobusnick commented on code/modules/wiremod/components/atom/remotecam.dm:
```diff
@@ -0,0 +1,438 @@
+ * Close the camera state (only if it's already active)
+ */
+/obj/item/circuit_component/remotecam/proc/close_camera()
+	if(shell_camera?.status)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/atom/remotecam.dm:
```diff
@@ -0,0 +1,438 @@
+		//Set camera network string
+		if(new_net_name)
+			shell_camera.network = list("[new_net_name]")
+		else
```
ранние ретурны

dageavtobusnick commented on code/modules/wiremod/components/atom/remotecam.dm:
```diff
@@ -0,0 +1,438 @@
+ */
+/obj/item/circuit_component/remotecam/proc/update_camera_location(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change)
+	SIGNAL_HANDLER
+	if(current_camera_state && current_cameranet_state)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/atom/remotecam.dm:
```diff
@@ -0,0 +1,438 @@
+	current_camera_emp = TRUE
+	close_camera()
+	current_camera_emp_timer_id = addtimer(CALLBACK(src, PROC_REF(remove_camera_emp)), REMOTECAM_EMP_RESET, TIMER_STOPPABLE)
+	for(var/mob/M as anything in GLOB.player_list)
```
имя переменной

dageavtobusnick commented on code/modules/wiremod/components/atom/remotecam.dm:
```diff
@@ -0,0 +1,438 @@
+	close_camera()
+	current_camera_emp_timer_id = addtimer(CALLBACK(src, PROC_REF(remove_camera_emp)), REMOTECAM_EMP_RESET, TIMER_STOPPABLE)
+	for(var/mob/M as anything in GLOB.player_list)
+		if(M.client?.eye == shell_camera)
```
ранний континюэ

dageavtobusnick commented on code/modules/wiremod/components/atom/remotecam.dm:
```diff
@@ -0,0 +1,438 @@
+	RegisterSignal(shell_parent, COMSIG_ORGAN_IMPLANTED, PROC_REF(on_organ_implanted))
+	RegisterSignal(shell_parent, COMSIG_ORGAN_REMOVED, PROC_REF(on_organ_removed))
+	var/obj/item/organ/internal/cyberimp/brain/bci/bci = shell_parent
+	if(bci.owner) //If somehow the camera was added while shell is already installed inside a mob, assign signals
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/atom/remotecam.dm:
```diff
@@ -0,0 +1,438 @@
+
+/obj/item/circuit_component/remotecam/polaroid/register_shell(atom/movable/shell)
+	. = ..()
+	if(!istype(shell_parent, /obj/item/camera))
```
изхелпер?

dageavtobusnick commented on code/modules/wiremod/components/atom/remotecam.dm:
```diff
@@ -0,0 +1,438 @@
+
+/obj/item/circuit_component/remotecam/airlock/register_shell(atom/movable/shell)
+	. = ..()
+	if(!istype(shell_parent, /obj/machinery/door/airlock))
```
изхелпер?

dageavtobusnick commented on code/modules/wiremod/components/atom/remotecam.dm:
```diff
@@ -0,0 +1,438 @@
+	if(!istype(shell_parent, /obj/machinery/door/airlock))
+		return
+	current_camera_state = FALSE //Always reset camera state for built-in shell components
+	shell_camera = new /obj/machinery/camera (shell_parent)
```
```suggestion
	shell_camera = new (shell_parent)
```

dageavtobusnick commented on code/modules/wiremod/components/atom/remotecam.dm:
```diff
@@ -0,0 +1,438 @@
+	if(!istype(shell_parent, /mob/living/simple_animal/circuit_drone))
+		return
+	current_camera_state = FALSE //Always reset camera state for built-in shell components
+	shell_camera = new /obj/machinery/camera (shell_parent)
```
```suggestion
	shell_camera = new (shell_parent)
```

dageavtobusnick commented on code/modules/wiremod/components/atom/remotecam.dm:
```diff
@@ -0,0 +1,438 @@
+	if(!istype(shell_parent, /obj/item/camera))
+		return
+	current_camera_state = FALSE //Always reset camera state for built-in shell components
+	shell_camera = new /obj/machinery/camera (shell_parent)
```
```suggestion
	shell_camera = new (shell_parent)
```

dageavtobusnick commented on code/modules/wiremod/components/atom/species.dm:
```diff
@@ -0,0 +1,32 @@
+	output = add_output_port("Раса", PORT_TYPE_STRING)
+
+/obj/item/circuit_component/species/input_received(datum/port/input/port)
+
```
зачем тут пустая строка?

dageavtobusnick commented on code/modules/wiremod/components/atom/species.dm:
```diff
@@ -0,0 +1,32 @@
+
+/obj/item/circuit_component/species/populate_ports()
+	input_port = add_input_port("Организм", PORT_TYPE_ATOM)
+
```
и тут

chomasuke replied:
читаемость. отделяем входы от выходов

dageavtobusnick commented on code/modules/wiremod/components/bci/hud/counter_overlay.dm:
```diff
@@ -0,0 +1,136 @@
+		)
+		number_alt_appearance.show_to(owner)
+
+		numbers += WEAKREF(number_alt_appearance)
```
слишком большой прок. Разбей

dageavtobusnick commented on code/modules/wiremod/components/bci/hud/object_overlay.dm:
```diff
@@ -0,0 +1,181 @@
+		show_to_owner(target_atom, owner)
+
+	var/datum/atom_hud/existing_overlay = LAZYACCESS(active_overlays, target_atom.UID())
+	if(COMPONENT_TRIGGERED_BY(signal_off, port) && !isnull(existing_overlay))
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/bci/hud/target_intercept.dm:
```diff
@@ -0,0 +1,69 @@
+	user.client.click_intercept = null
+	clicked_atom.set_output(object)
+	trigger_output.set_output(COMPONENT_SIGNAL)
+	if(parent.shell)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/bci/install_detector.dm:
```diff
@@ -0,0 +1,50 @@
+
+/obj/item/circuit_component/install_detector/register_shell(atom/movable/shell)
+	. = ..()
+	if(istype(shell, /obj/item/organ/internal/cyberimp/brain/bci))
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/bci/reagent_injector.dm:
```diff
@@ -0,0 +1,70 @@
+
+/obj/item/circuit_component/reagent_injector/register_shell(atom/movable/shell)
+	. = ..()
+	if(istype(shell, /obj/item/organ/internal/cyberimp/brain/bci))
```
ранний ретурн  и изхелпер

dageavtobusnick commented on code/modules/wiremod/components/bci/reagent_injector.dm:
```diff
@@ -0,0 +1,70 @@
+		bci = shell
+		bci.container_type = OPENCONTAINER
+		bci.create_reagents(15)
+		if(reagents.total_volume)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/bci/reagent_injector.dm:
```diff
@@ -0,0 +1,70 @@
+	var/fraction = min(transfer_amounts.value/bci.reagents.total_volume, 1)
+	bci.reagents.reaction(bci.owner, REAGENT_INGEST, fraction, ignore_protection = TRUE)
+	var/units = bci.reagents.trans_to(bci.owner, transfer_amounts.value)
+	if(units)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/bci/thought_listener.dm:
```diff
@@ -0,0 +1,65 @@
+	failure = add_output_port("Провал", PORT_TYPE_SIGNAL)
+
+/obj/item/circuit_component/thought_listener/register_shell(atom/movable/shell)
+	if(istype(shell, /obj/item/organ/internal/cyberimp/brain/bci))
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/bci/vox.dm:
```diff
@@ -0,0 +1,60 @@
+
+/obj/item/circuit_component/vox/pre_input_received(datum/port/input/port)
+	var/current_option = type_option.value
+	if(current_type != current_option)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/bci/vox.dm:
```diff
@@ -0,0 +1,60 @@
+
+		for(var/word in words_list)
+			play_vox_word(word, only_listener = owner)
+	else
```
ранний ретурн

dageavtobusnick reviewed (changes requested):
Часть 3. Тгуи надо доперевести + заменить content на children в кнопках. Так же обрати внимание на пустые строки в dm коде. Кое где они лишние.

dageavtobusnick commented on code/modules/wiremod/components/id/access_checker.dm:
```diff
@@ -0,0 +1,134 @@
+
+/obj/item/circuit_component/compare/access/ui_interact(mob/user, datum/tgui/ui)
+	ui = SStgui.try_update_ui(user, src, ui)
+	if(!ui)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/list/assoc_list_pick.dm:
```diff
@@ -0,0 +1,29 @@
+	input_list = add_input_port("Список", PORT_TYPE_ASSOC_LIST(PORT_TYPE_STRING, PORT_TYPE_ANY))
+
+/obj/item/circuit_component/list_pick/assoc/pre_input_received(datum/port/input/port)
+	if(port == list_options)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/list/assoc_list_pick.dm:
```diff
@@ -0,0 +1,29 @@
+	if(choice_list[choice])
+		output.set_output(choice_list[choice])
+		success.set_output(COMPONENT_SIGNAL)
+	else
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/list/assoc_list_set.dm:
```diff
@@ -0,0 +1,54 @@
+
+/obj/item/circuit_component/variable/assoc_list/list_set/pre_input_received(datum/port/input/port)
+	. = ..()
+	if(current_variable)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/list/concat.dm:
```diff
@@ -0,0 +1,45 @@
+	for(var/entry in list_input)
+		if(isdatum(entry))
+			text_list += PORT_TYPE_ATOM
+		else
```
ранний continue

dageavtobusnick commented on code/modules/wiremod/components/list/filter.dm:
```diff
@@ -0,0 +1,85 @@
+	list_options = add_option_port("Тип", GLOB.wiremod_basic_types)
+
+/obj/item/circuit_component/filter_list/pre_input_received(datum/port/input/port)
+	if(port == list_options)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/list/filter.dm:
```diff
@@ -0,0 +1,85 @@
+
+/obj/item/circuit_component/filter_list/proc/accept_entry_port(datum/port/input/port, list/return_values)
+	CIRCUIT_TRIGGER
+	if(return_values)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/list/filter.dm:
```diff
@@ -0,0 +1,85 @@
+			balloon_alert_to_viewers("начинает перегреваться!")
+			on_failed.set_output(COMPONENT_SIGNAL)
+			return
+		if(result["accept_entry"])
```
ранние континюэ

dageavtobusnick commented on code/modules/wiremod/components/list/foreach.dm:
```diff
@@ -0,0 +1,69 @@
+	list_options = add_option_port("Тип", GLOB.wiremod_basic_types)
+
+/obj/item/circuit_component/foreach/pre_input_received(datum/port/input/port)
+	if(port == list_options)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/list/format.dm:
```diff
@@ -0,0 +1,121 @@
+
+/obj/item/circuit_component/format/input_received(datum/port/input/port, list/return_values)
+	. = ..()
+
```
зачем?

dageavtobusnick commented on code/modules/wiremod/components/list/index.dm:
```diff
@@ -0,0 +1,71 @@
+	output = add_output_port("Результат", PORT_TYPE_ANY)
+
+/obj/item/circuit_component/index/pre_input_received(datum/port/input/port)
+	if(port == list_options)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/list/index.dm:
```diff
@@ -0,0 +1,71 @@
+	list_port = add_input_port("Список", PORT_TYPE_ASSOC_LIST(PORT_TYPE_STRING, PORT_TYPE_ANY))
+
+/obj/item/circuit_component/index/assoc_string/pre_input_received(datum/port/input/port)
+	if(port == list_options)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/list/list_add.dm:
```diff
@@ -0,0 +1,52 @@
+
+/obj/item/circuit_component/variable/list/listadd/pre_input_received(datum/port/input/port)
+	. = ..()
+	if(current_variable)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/list/list_add.dm:
```diff
@@ -0,0 +1,52 @@
+
+	if(!allow_duplicate.value)
+		info |= data_to_add
+	else
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/list/list_find.dm:
```diff
@@ -0,0 +1,64 @@
+	if(actual_result != 0)
+		result.set_output(TRUE)
+		found.set_output(COMPONENT_SIGNAL)
+	else
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/list/list_literal.dm:
```diff
@@ -0,0 +1,77 @@
+	list_options = add_option_port("Тип", GLOB.wiremod_basic_types)
+
+/obj/item/circuit_component/list_literal/pre_input_received(datum/port/input/port)
+	if(port == list_options)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/list/list_pick.dm:
```diff
@@ -0,0 +1,79 @@
+	if(choice)
+		output.set_output(choice)
+		success.set_output(COMPONENT_SIGNAL)
+	else
```
ранний ретурн. Вообщем во всем модуле типичные ошибки с ранним ретурном и континюэ. Поправь их. Больше не указываю.

dageavtobusnick commented on code/modules/wiremod/core/admin_panel.dm:
```diff
@@ -0,0 +1,76 @@
+
+	switch(action)
+		if("duplicate_circuit")
+			if(alert(usr, "Это создаст новую схему в том месте, где вы находитесь. Вы уверены?", "Подтверждение", "Да", "Нет") != "Да") // перевести на TGUI алёрт
```
тгуи алерт

dageavtobusnick commented on code/modules/wiremod/core/duplicator.dm:
```diff
@@ -0,0 +1,249 @@
+
+				port.connect(output_port)
+
+	SEND_SIGNAL(src, COMSIG_CIRCUIT_POST_LOAD)
```
Прок нужно разбить

dageavtobusnick commented on code/modules/wiremod/core/duplicator.dm:
```diff
@@ -0,0 +1,249 @@
+
+	SEND_SIGNAL(src, COMSIG_CIRCUIT_PRE_SAVE_TO_JSON, general_data)
+
+	return json_encode(general_data)
```
и этот

dageavtobusnick commented on code/modules/wiremod/core/duplicator.dm:
```diff
@@ -0,0 +1,249 @@
+		if("Файл")
+			txt = file2text(input(usr, "Укажите файл") as null|file)
+		if("Прямой ввод")
+			txt = input(usr, "Введите JSON-строку", "Прямой ввод") as text|null
```
тгуи инпут

chomasuke replied:
не стоит. первый инпут - это выбор файла. второй - ввод потенциально невероятно длинной строки

dageavtobusnick replied:
Хорошо, тогда текст замени на message, чтобы был мультилайн. Алерт чуть выше нужно сделать тгуи, в любом случае. Так же "Прямой ввод" и " Файл" нужно дефайнизировать.

dageavtobusnick replied:
Это тоже нужно сделать

> Хорошо, тогда текст замени на message, чтобы был мультилайн. Алерт чуть выше нужно сделать тгуи, в любом случае. Так же "Прямой ввод" и " Файл" нужно дефайнизировать.

dageavtobusnick commented on code/modules/wiremod/core/integrated_circuit.dm:
```diff
@@ -0,0 +1,808 @@
+		add_component_manually(I, user)
+		return
+
+	if(istype(I, /obj/item/stock_parts/cell))
```
изхелпер

dageavtobusnick commented on code/modules/wiremod/core/integrated_circuit.dm:
```diff
@@ -0,0 +1,808 @@
+
+/obj/item/integrated_circuit/attackby(obj/item/I, mob/living/user, params)
+	. = ..()
+	if(istype(I, /obj/item/circuit_component))
```
и тут

dageavtobusnick commented on code/modules/wiremod/core/usb_cable.dm:
```diff
@@ -0,0 +1,131 @@
+	ADD_TRAIT(src, TRAIT_NO_CLONE_IN_EXPERIMENTATOR, INNATE_TRAIT)
+	RegisterSignal(src, COMSIG_MOVABLE_MOVED, PROC_REF(on_moved))
+
+/obj/item/usb_cable/Destroy()
```
```suggestion
/obj/item/usb_cable/Destroy(force)
```

dageavtobusnick commented on code/modules/wiremod/datatypes/string.dm:
```diff
@@ -0,0 +1,17 @@
+	// So that they can't easily get the name like this.
+	if(isatom(value_to_convert))
+		return PORT_TYPE_ATOM
+	else
```
Зачем тут else?

dageavtobusnick commented on code/modules/wiremod/shell/brain_computer_interface.dm:
```diff
@@ -0,0 +1,617 @@
+/obj/machinery/bci_implanter/on_deconstruction(disassembled)
+	drop_stored_bci()
+
+/obj/machinery/bci_implanter/Destroy()
```
```suggestion
/obj/machinery/bci_implanter/Destroy(force)
```

dageavtobusnick commented on code/modules/wiremod/shell/brain_computer_interface.dm:
```diff
@@ -0,0 +1,617 @@
+
+/obj/machinery/bci_implanter/verb/eject()
+	set src in oview(1)
+	set name = "Eject occupant"
```
перевод

dageavtobusnick commented on code/modules/wiremod/shell/drone.dm:
```diff
@@ -0,0 +1,140 @@
+/mob/living/simple_animal/circuit_drone/examine(mob/user)
+	. = ..()
+	if(health < maxHealth)
+		if(health > maxHealth/3)
```
```suggestion
		if(health > maxHealth / 3)
```

dageavtobusnick reviewed (changes requested):
Очень многие из тайпы можно и нужно заменить изхелперами, создав их при необходимости. Перепроверь весь код и замени.

PlayerUnknown14 commented on code/modules/research/designs/wiremod_designs.dm:
Если мой перевод НИО будет замёржен раньше, нужно будет удалить неймы и дески у дизайнов + категории в датум.

PlayerUnknown14 committed:
Фиксы локализации + перевод интерфейсов

chomasuke committed:
чистить вилкой 05

coде ге\/!е\л/
Содержимое строки -> Вхождение подстроки
для nfc более не нужно ядро телекомов
фикс педальского списка плат
всей душой ненавижу ебаный tgui и ебаных childrenов

dageavtobusnick commented on code/_onclick/hud/radial.dm:
```diff
@@ -229,12 +253,20 @@ GLOBAL_LIST_EMPTY(radial_menus)
+		var/datum/radial_menu_choice/choice = to_extract_from
+		to_extract_from = choice.image
+
+	var/mutable_appearance/MA = new /mutable_appearance(to_extract_from)
```
Имя переменной

dageavtobusnick commented on code/datums/components/material_container.dm:
```diff
@@ -331,6 +331,25 @@
+/datum/component/material_container/ui_data(mob/user)
+	var/list/data = list()
+
+	for(var/MAT in materials)
```
Имя переменной

dageavtobusnick commented on code/datums/components/usb_port.dm:
```diff
@@ -0,0 +1,262 @@
+
+/datum/component/usb_port/proc/set_physical_object(atom/movable/new_physical_object)
+	if(physical_object)
+		unregister_physical_signals()
```
пустую строку

dageavtobusnick commented on tgui/packages/tgui/interfaces/Reflector.tsx:
```diff
@@ -0,0 +1,252 @@
+                      fluid
+                      icon="redo-alt"
+                      iconPosition="right"
+                      content="+15"
```
Как я и сказал, весь content в кнопках нужно перевести в children.

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,137 +1,208 @@
+/datum/atom_hud/alternate_appearance/proc/apply_to_new_mob(mob/applying_to)
+	if(!mob_should_see(applying_to))
+		return FALSE
+	if(!hud_users_all_z_levels[applying_to])
```
пустые строки

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,137 +1,208 @@
+		return
+	if(alternate_appearances && alternate_appearances[key])
+		return
+	if(!ispath(type, /datum/atom_hud/alternate_appearance))
```
пустые строки

dageavtobusnick commented on code/game/alternate_appearance.dm:
```diff
@@ -1,137 +1,208 @@
 		return
-	var/datum/alternate_appearance/AA = alternate_appearances[key]
-	if(!AA || !AA.img)
+	untrack_mob(former_viewer)
```
пустую строку

dageavtobusnick commented on code/game/atoms.dm:
```diff
@@ -714,6 +711,9 @@
 /atom/proc/tool_act(mob/living/user, obj/item/I, tool_type)
+	var/signal_result = SEND_SIGNAL(src, COMSIG_ATOM_TOOL_ACT(tool_type), user, I)
+	if(signal_result)
+		return TRUE
```
пустые строки

dageavtobusnick commented on code/game/machinery/rechargestation.dm:
```diff
@@ -242,51 +243,48 @@
 		can_accept_user = 1
 
+	if(is_circuit_drone(user))
+		can_accept_user = 1
```
```suggestion
		can_accept_user =  TRUE
```

dageavtobusnick commented on code/modules/atmospherics/machinery/components/binary_devices/pump.dm:
```diff
@@ -72,13 +84,16 @@ Thus, the two variables affect pump operation are set in New():
+		return
+
+	underlays.Cut()
+	var/turf/T = get_turf(src)
```
Имя перменной

dageavtobusnick commented on code/modules/mob/living/carbon/human/inventory.dm:
```diff
@@ -586,6 +586,13 @@
+	var/list/visible_items = ..()
+	var/obj/item/clothing/under/under = w_uniform
+	if(istype(under) && length(under.accessories) && (under in visible_items))
+		visible_items += under.accessories
```
пустые строки

dageavtobusnick commented on code/modules/mob/living/carbon/inventory.dm:
```diff
@@ -390,6 +390,38 @@
+		if(equipped)
+			visible_items += equipped
+	for(var/obj/item/held in get_equipped_items(INCLUDE_HELD))
+		visible_items += held
```
отдели ифы и форы пустыми строками

dageavtobusnick commented on code/modules/paperwork/photography/video_camera.dm:
```diff
@@ -61,19 +61,29 @@ GLOBAL_LIST_EMPTY(active_video_cameras)
+	if(get_dist(src, M) <= canhear_range)
+		talk_into(M, msg)
+
+	for(var/obj/machinery/computer/security/telescreen/T in SSmachines.get_by_type(/obj/machinery/computer/security/telescreen))
```
Имя переменной тут и ниже

dageavtobusnick commented on code/modules/wiremod/components/abstract/module.dm:
```diff
@@ -0,0 +1,347 @@
+/obj/item/circuit_component/module/pre_input_received(datum/port/input/port)
+	if(!port)
+		return
+	var/datum/port/output/port_to_update = linked_ports[port]
```
пустая строка

dageavtobusnick commented on code/modules/wiremod/components/abstract/module.dm:
```diff
@@ -0,0 +1,347 @@
+			remove_output_port(output_ports[port_id])
+			output_component.remove_input_port(removed_port)
+			. = TRUE
+		if("set_port_name", "set_port_type")
```
пустыми строками отдели ифы во всем проке

dageavtobusnick commented on code/modules/wiremod/components/action/pathfind.dm:
```diff
@@ -0,0 +1,146 @@
+		access = id.GetAccess()
+		return access
+
+	if(id_card.value)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/atom/health_state.dm:
```diff
@@ -0,0 +1,64 @@
+			return organism.InCritical()
+		if(UNCONS)
+			return state == UNCONS
+		if(DECEASED)
```
ифы в свиче пустыми строками отдели

dageavtobusnick commented on code/modules/wiremod/components/atom/remotecam.dm:
```diff
@@ -0,0 +1,494 @@
+	UnregisterSignal(shell_parent, COMSIG_ATOM_EMP_ACT)
+	if(current_camera_emp)
+		deltimer(current_camera_emp_timer_id)
+		current_camera_emp = FALSE
```
отдели пустыми строками ифы

dageavtobusnick commented on code/modules/wiremod/components/list/list_literal.dm:
```diff
@@ -0,0 +1,83 @@
+		if(lists > max_list_count)
+			return lists
+
+	return lists
```
Это по-хорошему перенести в хелперы листов

dageavtobusnick commented on code/modules/wiremod/components/math/arithmetic.dm:
```diff
@@ -0,0 +1,99 @@
+				result = max(result, value)
+			if(COMP_ARITHMETIC_MIN)
+				result = min(result, value)
+			if(COMP_ARITHMETIC_EXPONENTIATION)
```
отдели пустыми строками ифы в свиче

dageavtobusnick commented on code/modules/wiremod/components/utility/nfc_send.dm:
```diff
@@ -0,0 +1,44 @@
+		data_package.set_datatype(PORT_TYPE_LIST(new_datatype))
+
+/obj/item/circuit_component/nfc_send/input_received(datum/port/input/port)
+	if(isatom(target.value))
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/utility/nfc_send.dm:
```diff
@@ -0,0 +1,44 @@
+	target = add_input_port("Цель", PORT_TYPE_ATOM)
+
+/obj/item/circuit_component/nfc_send/pre_input_received(datum/port/input/port)
+	if(port == list_options)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/utility/nfc_send_literal.dm:
```diff
@@ -0,0 +1,32 @@
+
+/obj/item/circuit_component/list_literal/nfc_send/input_received(datum/port/input/port)
+	. = ..()
+	if(isatom(target.value))
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/variables/getter.dm:
```diff
@@ -0,0 +1,21 @@
+
+/obj/item/circuit_component/variable/getter/pre_input_received(datum/port/input/port)
+	. = ..()
+	if(current_variable)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/components/variables/setter.dm:
```diff
@@ -0,0 +1,34 @@
+
+/obj/item/circuit_component/variable/setter/pre_input_received(datum/port/input/port)
+	. = ..()
+	if(port == variable_name)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/core/component.dm:
```diff
@@ -0,0 +1,485 @@
+	ADD_TRAIT(src, TRAIT_NO_CLONE_IN_EXPERIMENTATOR, INNATE_TRAIT)
+	if((circuit_flags & CIRCUIT_FLAG_INPUT_SIGNAL) && !trigger_input)
+		trigger_input = add_input_port("Вызов", PORT_TYPE_SIGNAL, order = 2)
+	if((circuit_flags & CIRCUIT_FLAG_OUTPUT_SIGNAL) && !trigger_output)
```
пустые строки

dageavtobusnick commented on code/modules/wiremod/core/component.dm:
```diff
@@ -0,0 +1,485 @@
+		ui_alerts -= "tachometer-alt"
+	if(circuit_flags & CIRCUIT_FLAG_DISABLED)
+		ui_alerts["exclamation"] = "Нефункционально"
+	else
```
и в этом проке

dageavtobusnick commented on code/modules/wiremod/core/integrated_circuit.dm:
```diff
@@ -0,0 +1,811 @@
+		owner_id = WEAKREF(tool)
+		return
+
+	if(tool.tool_behaviour == TOOL_SCREWDRIVER)
```
ранний ретурн

dageavtobusnick commented on code/modules/wiremod/core/integrated_circuit.dm:
```diff
@@ -0,0 +1,811 @@
+		return
+	shell.name = initial(shell.name)
+	for(var/obj/item/circuit_component/attached_component as anything in attached_components)
+		attached_component.unregister_shell(shell)
```
пустые строки

dageavtobusnick commented on code/modules/wiremod/core/integrated_circuit.dm:
```diff
@@ -0,0 +1,811 @@
+				if(module_component == to_check)
+					continue
+				if(istype(module_component, to_check.type))
+					return TRUE
```
пустые строки

dageavtobusnick commented on code/modules/wiremod/core/integrated_circuit.dm:
```diff
@@ -0,0 +1,811 @@
+	.["examined_rel_y"] = examined_rel_y
+	.["grid_mode"] = grid_mode
+
+	.["is_admin"] = (admin_only || user.can_advanced_admin_interact()) && check_rights_for(user.client, R_VAREDIT)
```
тут тоже

dageavtobusnick commented on code/modules/wiremod/core/integrated_circuit.dm:
```diff
@@ -0,0 +1,811 @@
+				return
+			component.rel_x = text2num(params["rel_x"])
+			component.rel_y = text2num(params["rel_y"])
+			return TRUE
```
отдели ифы пустыми строками во всем проке

dageavtobusnick commented on code/modules/wiremod/core/integrated_circuit.dm:
```diff
@@ -0,0 +1,811 @@
+	SIGNAL_HANDLER
+	// This'll also be called in the Destroy() override of /obj/item/circuit_component
+	if(!QDELING(source))
+		qdel(source)
```
пустая строка

dageavtobusnick commented on code/modules/wiremod/core/port.dm:
```diff
@@ -0,0 +1,244 @@
+/datum/port/New(obj/item/circuit_component/to_connect, name, datatype, order = 1)
+	if(!to_connect)
+		qdel(src)
+		return
```
пустая строка

dageavtobusnick commented on code/modules/wiremod/core/port.dm:
```diff
@@ -0,0 +1,244 @@
+		else
+			src.value = datatype_handler.convert_value(src, value, force)
+		if(isdatum(value))
+			RegisterSignal(value, COMSIG_QDELETING, PROC_REF(null_value))
```
пустые строки

dageavtobusnick commented on code/modules/wiremod/dcs_components/component_add_port.dm:
```diff
@@ -0,0 +1,92 @@
+		if(is_output)
+			component.remove_output_port(pop(port_list))
+		else
+			component.remove_input_port(pop(port_list))
```
пустые строки

dageavtobusnick commented on code/modules/wiremod/shell/brain_computer_interface.dm:
```diff
@@ -0,0 +1,622 @@
+		to_chat(user, span_warning("[target] не помест[PLUR_IT_YAT(target)]ся в [declent_ru(ACCUSATIVE)], пока на [GEND_ON_IN_HIM(target)] сидит слайм!"))
+		return TRUE
+	put_in(target, user)
+	return TRUE
```
пустые строки во всем проке

dageavtobusnick commented on code/modules/wiremod/shell/brain_computer_interface.dm:
```diff
@@ -0,0 +1,622 @@
+		return
+	if(user.loc == null) // just in case someone manages to get a closet into the blue light dimension, as unlikely as that seems
+		return
+	if(!istype(user.loc, /turf) || !istype(target.loc, /turf)) // are you in a container/closet/pod/etc?
```
изхелперы

dageavtobusnick commented on code/modules/wiremod/shell/brain_computer_interface.dm:
```diff
@@ -0,0 +1,622 @@
+		to_chat(grabber, span_warning("[target] не помест[PLUR_IT_YAT(target)]ся в [declent_ru(ACCUSATIVE)], пока на [GEND_ON_IN_HIM(target)] сидит слайм!"))
+		return .
+	put_in(target, grabber)
+	add_fingerprint(grabber)
```
пустые строки

dageavtobusnick commented on code/modules/wiremod/shell/dispenser.dm:
```diff
@@ -0,0 +1,217 @@
+			add_item(user, bag_item)
+		return ATTACK_CHAIN_BLOCKED
+	add_item(user, item)
+	return ATTACK_CHAIN_BLOCKED
```
пустые строки

dageavtobusnick commented on code/modules/wiremod/shell/dispenser.dm:
```diff
@@ -0,0 +1,217 @@
+
+/obj/structure/dispenser_bot/wrench_act(mob/living/user, obj/item/tool)
+	if(locked)
+		return
```
пустая строка

dageavtobusnick reviewed (changes requested):
Думаю после этого можно мерджить

chomasuke committed:
чистить вилкой 0сколько?
